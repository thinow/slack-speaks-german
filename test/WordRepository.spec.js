const WordRepository = require('../src/WordRepository')
const WordRepositoryFolderError = require('../src/errors/WordRepositoryFolderError')

describe('when loading folder', () => {

    describe('used in production', () => {
        it('should successfully load the repository', () => {
            // when
            const repository = WordRepository.loadFromFolder('./resources/words')

            // then
            expect(repository).toBeInstanceOf(WordRepository)
        })
    })

    describe('which is not a folder', () => {
        it('should throw an error', () => {
            expect(() => {
                WordRepository.loadFromFolder('Not a folder')
            }).toThrow(WordRepositoryFolderError)
        })
    })

    describe('which does not contain data.txt', () => {
        it('should throw an error', () => {
            expect(() => {
                WordRepository.loadFromFolder('./test/resources/folder-without-data')
            }).toThrow(WordRepositoryFolderError)
        })
    })

    describe('which does not contain metadata.json', () => {
        it('should throw an error', () => {
            expect(() => {
                WordRepository.loadFromFolder('./test/resources/folder-without-metadata')
            }).toThrow(WordRepositoryFolderError)
        })
    })
})


describe('with a single word', () => {
    it('should return the word as the first item', async () => {
        // when
        const repository = WordRepository.loadFromFolder('./test/resources/folder-single-word');

        // then
        expect(repository.getNumberOfWords()).toEqual(1)
        expect(await repository.getWord(0)).toEqual(expect.objectContaining(({
            german: 'einzig',
            english: 'only, unique',
        })))
    })
})

describe('with multiple words', () => {
    it('should the word based on the index', async () => {
        // when
        const repository = WordRepository.loadFromFolder('./test/resources/folder-multiple-words');

        // then
        expect(await repository.getWord(1)).toEqual(expect.objectContaining(({
            german: 'erz√§hlen',
            english: 'to tell',
        })))
    })
})
