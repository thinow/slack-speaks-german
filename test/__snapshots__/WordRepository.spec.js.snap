// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with multiple words should correctly parse all the words based on the index 1`] = `
Word {
  "details": "no plural forms",
  "english": "replacement",
  "examples": Array [
    Object {
      "english": "You get from our company a replacement for the broken device.",
      "german": "Sie bekommen von unserer Firma einen Ersatz für das kaputte Gerät.",
    },
  ],
  "german": "der Ersatz",
  "germanWordOnly": "Ersatz",
}
`;

exports[`with multiple words should correctly parse all the words based on the index 2`] = `
Word {
  "details": "erzählt, erzählte, hat erzählt",
  "english": "to tell",
  "examples": Array [
    Object {
      "english": "My husband has already told me a lot about you.",
      "german": "Mein Mann hat mir schon viel von Ihnen erzählt.",
    },
    Object {
      "english": "I always have to tell my children a story in the evening.",
      "german": "Ich muss meinen Kindern abends immer eine Geschichte erzählen.",
    },
  ],
  "german": "erzählen",
  "germanWordOnly": "erzählen",
}
`;

exports[`with multiple words should correctly parse all the words based on the index 3`] = `
Word {
  "details": "",
  "english": "due",
  "examples": Array [
    Object {
      "english": "The bill is due at the end of the month.",
      "german": "Die Rechnung ist Ende des Monats fällig.",
    },
  ],
  "german": "fällig",
  "germanWordOnly": "fällig",
}
`;
